### PR TITLE
Mpi

### PR DESCRIPTION
**Pull request by TApplencourt**

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -100,17 +100,13 @@ AC_CONFIG_FILES([Makefile
...
 AC_CONFIG_FILES([ze/tracer_ze.sh], [chmod +x ze/tracer_ze.sh])
-AC_CONFIG_FILES([ze/babeltrace_ze], [chmod +x ze/babeltrace_ze])
 AC_CONFIG_FILES([ze/test_wrapper.sh], [chmod +x ze/test_wrapper.sh])
-AC_CONFIG_FILES([cuda/babeltrace_cuda], [chmod +x cuda/babeltrace_cuda])
 AC_CONFIG_FILES([cuda/tracer_cuda.sh], [chmod +x cuda/tracer_cuda.sh])
 AC_CONFIG_FILES([cuda/test_wrapper.sh], [chmod +x cuda/test_wrapper.sh])
 AC_CONFIG_FILES([xprof/test_wrapper.sh], [chmod +x xprof/test_wrapper.sh])
 AC_CONFIG_FILES([utils/babeltrace_thapi], [chmod +x utils/babeltrace_thapi])
 AC_CONFIG_FILES([omp/tracer_omp.sh], [chmod +x omp/tracer_omp.sh])
-AC_CONFIG_FILES([omp/babeltrace_omp], [chmod +x omp/babeltrace_omp])
```
Comment (Kerilk) [inline configure.ac:115]: Please remove the concerned ".in" files from the repository as well.

File: utils/babeltrace_thapi.in [hunk 2]
```diff
@@ -1,118 +1,194 @@
...
+                 opts.on("--#{k}=VALUE", t) do |v|
+                   $options_tally[k] = [t, v]
+                 end
+               end
+               common_options(opts)
+             end,
+  'timeline' => OptionParser.new do |opts|
+                  opts.banner = 'Usage: timeline [OPTIONS] trace_directory...'
+                  common_options(opts)
+                end
```
Comment (Kerilk) [inline utils/babeltrace_thapi.in:77]: Fix indentation here.

File: utils/babeltrace_thapi.in [hunk 3]
```diff
@@ -1,118 +1,194 @@
...
+                    FileTest.directory?(path)
+                  end.select do |path|
+                    File.basename(path) == 'metadata'
+                  end.collect do |path|
+                    File.dirname(path)
+                  end.select do |path|
+                    qe = BT2::BTQueryExecutor.new(component_class: ctf_fs, object_name: 'babeltrace.support-info',
+                                                  params: { 'input' => path, 'type' => 'directory' })
+                    qe.query.value['weight'] > 0.5
+                  end
```
Comment (Kerilk) [inline utils/babeltrace_thapi.in:115]: Fix indentation here.

File: utils/babeltrace_thapi.in [hunk 4]
```diff
@@ -1,118 +1,194 @@
...
-        str << " - #{e.name}: "
-        if l
-          str << l.call(e.payload_field.value)
-        else
-          str << e.payload_field.to_s
-        end
-        puts str
+      next unless l || !$options[:restrict]
+      str = "#{Time.at(0, m.get_default_clock_snapshot.ns_from_origin, :nsec).strftime('%H:%M:%S.%9L')}"
+      str << " - #{e.stream.trace.get_environment_entry_value_by_name('hostname')}" if $options[:context]
```
Comment (Kerilk) [inline utils/babeltrace_thapi.in:153]: Should go in the if block below.

File: utils/babeltrace_thapi.in (hunk 2 above)
Comment (TApplencourt) [reply utils/babeltrace_thapi.in:77]: how much intend? they are inside a dict, so 2 or 4 or 6?

File: utils/babeltrace_thapi.in (hunk 3 above)
Comment (TApplencourt) [reply utils/babeltrace_thapi.in]: Done

File: utils/babeltrace_thapi.in [hunk 5]
```diff
@@ -1,118 +1,194 @@
...
-  mess = iterator.next_messages
-  mess.each { |m|
-    if m.type == :BT_MESSAGE_TYPE_EVENT
+# Muxer
+comp_muxer = graph.add_component(utils_muxer, 'mux')
+
+# Other Comps
+serial_comps = 
+case command
+in 'trace'
```
Comment (Kerilk) [inline utils/babeltrace_thapi.in:138]:
```suggestion
when 'trace'
```
No reason to use pattern matching here, this is a regular case/when

File: utils/babeltrace_thapi.in [hunk 6]
```diff
@@ -1,118 +1,194 @@
...
+        end.join(', ')
       end
+      str << " - #{e.name}: "
+      str << (l ? l.call(e.payload_field.value) : e.payload_field.to_s)
+      puts str
     end
   }
-}
+  graph.add_simple_sink('babeltrace_thapi', consume)
+in 'tally'
```
Comment (Kerilk) [inline utils/babeltrace_thapi.in:166]:
```suggestion
when 'tally'
```

File: utils/babeltrace_thapi.in [hunk 7]
```diff
@@ -1,118 +1,194 @@
...
+  comp_intervals = interval_plugins.map.with_index do |comp, i|
+    graph.add_component(comp, "interval_#{i}")
+  end
+  comp_tally = graph.add_component(sink_plugin, 'tally',
+                                   params: $options_tally.transform_values { |_, v| v } )
+  [comp_intervals, comp_tally]
+in 'timeline'
```
Comment (Kerilk) [inline utils/babeltrace_thapi.in:173]:
```suggestion
when 'timeline'
```

File: utils/babeltrace_thapi.in (hunk 2 above)
Comment (Kerilk) [reply utils/babeltrace_thapi.in:77]: I would use the same euristic as for the =:
```ruby
  'trace' =>
    OptionParser.new do |opts|
    end
  'tally' =>
```

File: utils/babeltrace_thapi.in (hunk 4 above)
Comment (TApplencourt) [reply utils/babeltrace_thapi.in]: Indeed!

File: utils/babeltrace_thapi.in (hunk 2 above)
Comment (TApplencourt) [reply utils/babeltrace_thapi.in]: Good idea!

File: configure.ac (hunk 1 above)
Comment (Kerilk) [inline configure.ac:115]: Please fix scripts depending on these (tracers).
Putting it there so I don't forget on Monday...

Comment (TApplencourt) [reply configure.ac:115]: Done

**Review by Kerilk — APPROVED** [review]

**Pull request merged**